host/xtest/regression_1000.c

```diff
@@ -1381,24 +1386,51 @@ static void xtest_tee_test_1017(ADBG_Case_t *c)
 	TEEC_ReleaseSharedMemory(&shm);
 }
 
+static void invoke_1byte_out_of_bounds(ADBG_Case_t *c, TEEC_Session *session,
+				      TEEC_SharedMemory *shm)
+{
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_PARTIAL_INPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	op.params[0].memref.parent = shm;
+	op.params[0].memref.size = shm->size / 2;
+	op.params[0].memref.offset = shm->size - (shm->size / 2) + 1;
+
+	/*
+	 * Since reference are not valid the invocation request should NOT
+	 * reach the TA whatever the status. Yet we expect noboby gets hurt.
+	 */
+	ADBG_EXPECT_NOT(c, TEEC_ERROR_TARGET_DEAD,
+			TEEC_InvokeCommand(session, TA_OS_TEST_CMD_PARAMS, &op,
+					   &ret_orig));
+
+	ADBG_EXPECT_NOT(c, TEEC_ORIGIN_TRUSTED_APP, ret_orig);
```

Comment:
`TEEC_ORIGIN_COMMS` is expected

Reply:
I thin libteec should try to only do the sanity checks that aren't done in the kernel. For two reasons:
- both avoid cluttering the code with redundant checks
- enable us to check that the kernel does the expected checks or there will be a real security problem.